runner/lib/helpers/buffer-line-transform.js

```diff
@@ -0,0 +1,95 @@
+/* global Buffer */
+/* eslint-disable no-underscore-dangle,no-nested-ternary,no-plusplus */
+
+import { Transform } from 'stream';
+
+export default class BufferLineTransform extends Transform {
+  /**
+   * The BufferLineTransform is reading String or Buffer content from a Readable stream
+   * and writing each line as a Buffer in object mode
+   *
+   * @param {import('./buffer-line-transform.js').BufferLineTransformOptions} [options]
+   */
+  constructor(options) {
+    const { break: breakValue, breakEncoding, ...transformOptions } =
+      options || {};
+    super({ ...transformOptions, readableObjectMode: true });
+    this._breakValue = breakValue || 10;
+    this._breakEncoding = breakEncoding;
+
+    /** @type {number} */
+    let breakLength;
+    if (!breakValue || typeof breakValue === 'number') {
+      breakLength = 1;
+    } else if (Buffer.isBuffer(breakValue)) {
+      breakLength = breakValue.length;
+    } else {
+      breakLength = Buffer.from(breakValue, breakEncoding).length;
+    }
+    this._breakLength = breakLength;
```

Comment:
It would probably be simpler to normalize the break to a single type, or even defer that responsibility to the user.

Reply:
Deferring is exactly what this transform does, as it passes through to [`buf.indexOf()`](https://nodejs.org/docs/latest-v14.x/api/buffer.html#buffer_buf_indexof_value_byteoffset_encoding). However I do need to advance the search by the number of bytes of what had been found, which is what this calculates.